go/interpreter/lfvm/instructions.go

```diff
@@ -1080,6 +1077,19 @@ func genericCall(c *context, kind tosca.CallKind) {
 	}
 
 	baseGas := c.memory.ExpansionCosts(needed_memory_size)
+
+	accessCost := tosca.Gas(0)
+	if c.isAtLeast(tosca.R09_Berlin) {
```

Comment:
Because this complete block relates to a definition that combines two predicates, I would write it like:
```
// comment here
if c.isAtLeast(tosca.R09_Berlin) &&
  c.context.AccessAccount(toAddr) != WarmAccess{   
      accessCost = ColdAccountAccessCostEIP2929 - WarmStorageReadCostEIP2929
}
```  

Even more:
The duality of having a boolean type and two declared constants for it is an antipattern and the whole thing should be renamed, one of the two following:

- AccessAccount -> isWarmAccess (addr) bool
or
- AccessStatus stops being a bool, and becomes some kind of enum instead. 

```

Reply:
hopefully made the comparison clearer by using  `c.context.AccessAccount(toAddr) == tosca.ColdAccess`